iree/schemas/cuda_executable_def.fbs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace iree;
+
+// 'Cuda Executable'.
+file_identifier "CUDA";
+file_extension "cuda";
+
+// A struct for the kernel block size along each dimensions.
+struct block_size {
+  x:uint32;
+  y:uint32;
+  z:uint32;
+}
+
+table CudaExecutableDef {
+  // A map of entry point ordinals to string names as used in the shader
+  // library.
+  entry_points:[string];
+
+  // Block sizes for each entry point.
```

Comment:
note: one day it'd be nice to remove this limitation by making hal.command_buffer.dispatch take the local workgroup size at runtime, but even when we have that it's also worth (optionally) having here as you can use CU_JIT_THREADS_PER_BLOCK to pass this to cuModuleLoadDataEx and let it do the same kind of optimizations vulkan is doing (able to constant prop the values, etc).